### PR TITLE
feat(datastore): support merging local file with memory table

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

support merging local file with memory table

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/data_store.py [hunk 1]
```diff
@@ -34,6 +34,8 @@
 import requests
 import jsonlines
 from loguru import logger
+from filelock import FileLock
```
Comment (tianweidut) [inline client/starwhale/api/_impl/data_store.py:37]: add `filelock` pip requirement package into the setup.py ?

**Comment by tianweidut** [comment]
LGTM

File: client/starwhale/api/_impl/data_store.py (hunk 1 above)
Comment (jialeicui) [reply client/starwhale/api/_impl/data_store.py:37]: done

File: client/starwhale/api/_impl/data_store.py [hunk 2]
```diff
@@ -903,26 +883,48 @@ def loads(cls, file: Path, table_name: str) -> MemoryTable:
...
-            time.sleep(0.1)
+            for k, ir in self.records.items():
+                if k in dumped_keys:
+                    continue
+                writer.write(ir.dumps())
 
+        os.rename(temp_filename, dst)
         self.dirty = False
 
+    def _dump_from_local_file(self, file: Path, output: Writer) -> List[str]:
```
Comment (xuchuan) [inline client/starwhale/api/_impl/data_store.py:913]: ->Set[str] will be more efficient

**Review by xuchuan — CHANGES_REQUESTED** [review]

File: client/starwhale/api/_impl/data_store.py (hunk 2 above)
Comment (jialeicui) [reply client/starwhale/api/_impl/data_store.py]: done

**Review by tianweidut — APPROVED** [review]

**Pull request merged**